### ISSUE TITLE
Error after serving HTTP request over TLS

### ISSUE THREAD

**Issue by casey** [FlorianUekermann/rustls-acme#15]
Thanks for rustls-acme! We're using it to automatically fetch TLS certs for a daemon, and it's working great.

We're using the low-level interface, because we're using `tokio`, so we're configuring `ResolvesServerCertUsingAcme` but otherwise doing everything ourselves.

You can see our code [here](https://github.com/agora-org/agora/blob/a41d6250e2dce5d87b1ad2a4d966aa218affe99e/src/https_request_handler.rs#L67).

Fetching certs and serving connections is working perfectly, but the result of `Http::new().serve_connection(tls_stream, request_handler).await` is returning an error.

You can see the `log::error!` that's firing [here](https://github.com/agora-org/agora/blob/a41d6250e2dce5d87b1ad2a4d966aa218affe99e/src/https_request_handler.rs#L106).

The specific error message is:

```
[2021-10-20T22:22:10Z ERROR agora::https_request_handler] Error closing TLS connection: connection error: not connected
```

This isn't causing any problems for us, although we'd like to be able to stop ignoring this error, in case it masks more serious errors in the future, so we wanted to ask if you've seen this error before, or if you had any ideas as to what might be the problem.

Our TLS knowledge is pretty fuzzy, and it seems likely that this actually isn't a `rustls-acme` issue, but maybe an issue with our code, or with a lower-level crate, like `rustls` or `hyper`, but we thought we'd start with an issue here.

**Comment by FlorianUekermann** [comment]
I had a quick look at your code and the way you use the acceptor looks fine at first glance. I couldn't find your actual handler code. The error sounds like either the remote peer or server closed the connection, but someone is still doing stuff on it.

If you explicitly close the connection somewhere to clean up, maybe want to ignore the result.

This is interesting timing, because I just released 0.2.0, which removes the option from the return value of `accept` and hands you closed tls connections when it handled a let's encrypt connection. So I doubt that this error currently originates from rustls-acme, but in the future you will see errors like that during certificate refreshes (so very rarely).

If you point me to the actual handler code that is called after you get a tls connection, I can take another look if I see something obvious.

**Comment by casey** [comment]
> I had a quick look at your code and the way you use the acceptor looks fine at first glance.

Thanks for taking a look!

> The error sounds like either the remote peer or server closed the connection and then you still do stuff on it.

That seems reasonable, we thought of rustls-acme since we don't seem to be getting this error otherwise. However, we're not calling our handler the same way here, so it could easily be something else we're doing that's weird.

> If you point me to the actual handler code that is called after you get a tls connection, I can take another look if I see something obvious.

Our request handler's service implementation is [here](https://github.com/agora-org/agora/blob/ba1d99805e54718de832a86d69f1674084c0deaa/src/request_handler.rs#L97). Let us know if it looks weird, and otherwise we'll look elsewhere in the stack.

Thanks for the help!
